### DISCUSSION TITLE
Error while installing PyTorch geometric

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#6848] [Q&A] [answered]
I am using `M1 MacBook Air`. I am trying to install torch_geometric in new Conda environment. I installed `torch 1.13.1` along with `python 3.9.6`. As mentioned on official website of torch_geometric I selected options and I got following link `pip install pyg_lib torch_scatter torch_sparse torch_cluster torch_spline_conv torch_geometric -f https://data.pyg.org/whl/torch-1.13.0+cpu.html`.
I just changed `pip to pip3` in that command and used it and got following error
```
Defaulting to user installation because normal site-packages is not writeable
Looking in links: https://data.pyg.org/whl/torch-1.13.0+cpu.html
ERROR: Could not find a version that satisfies the requirement pyg_lib (from versions: none)
ERROR: No matching distribution found for pyg_lib
```

Can anyone help me in this.

**Comment by FinAminToastCrunch** [comment]
Same issue but with windows and Cuda 11.6 and pytorch 1.13

**Reply by FinAminToastCrunch** [reply]
Edit: I think that the pip command from this website should be followed: https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html

The github pip command is wrong.

**Reply by utkarsh0902311047** [reply]
I am still facing this error. Did you solve this problem?

**Reply by FinAminToastCrunch** [reply]
As odd as it sounds, downgrading my cuda version to 11.2 somehow fixed it.

I am not sure why or how this worked and I can not promise that it will work for you.

**Reply by utkarsh0902311047** [reply]
I am simply trying it for cpu

**Reply by MINJIK01** [reply]
Thanks! I think it works.

**Comment by utkarsh0902311047** [comment]
I am still facing this error. Does anyone solve this problem?

Initially I get the error 'NeigborSampler' required either pyg-lib or torch-sparse
I already have torch-sparse installed. When I try to install pyg-lib I keep getting the error.

**Comment by utkarsh0902311047** [comment]
Ok, I searched in issues and find out that need to pip install torch_scatter

**Reply by RushilRandhar** [reply]
I am having this issue. Can you elaborate on your solution. I tried that and it didn't work.

**Comment by atale480** [comment, marked as answer]
Recently I installed torch 2.0.0 and again repeated same process mentioned on website and all worked just fine.

**Comment by AndyJZhao** [comment]
I solved this via removing the `pyg-lib` in the pip installation command: 
`pip install torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-2.0.0+cpu.html`

**Reply by thboon3** [reply]
Worked for me, thanks! Ofc I just changed the command to fit my system: 
`pip install torch_scatter torch_sparse torch_cluster torch_spline_conv -f https://data.pyg.org/whl/torch-2.2.1+cu121.html
`

**Comment by Rayan-Allali** [comment]
Hello, is there a way to use torch_geometric with these torch==1.8.1+cu111?
